### PR TITLE
add `nbCodeBlock` to emit code in a `block` statement

### PR DESCRIPTION
**Pull request by Vindaar**
I started writing some first tutorials for `getting-started`. @HugoGranstrom gave me some feedback that for reading purposes it's better to only reference variables defined in the current code block.

With that in mind I realized that it'd be nice to just emit code into a `block` statement so that one doesn't have to write variables `x1` to `xN`. 

I added a `nbCodeBlock` (which I realized is the name internally used as well :see_no_evil:) and a static flag that emits it into a `block` in that case. Seems to work as I expected. So:

```nim
nbCodeBlock:
  let x = 5
nbCodeBlock:
  let x = 10
```
is valid, whereas of course
```nim
nbCode: 
  let x = 5
nbCode:
  let x = 10
```
still throws a redefinition error.

### CODE REVIEW

**Comment by pietroppeter** [comment]
I would actually go with:
```nim
block:
  nbCode:
    let x = 1
block:
  nbCode:
    let x = 2
```

**Comment by Vindaar** [comment]
Yes, Hugo mentioned that afterwards as well. Seems like a common use case that could use an overload, don't you agree?

**Comment by HugoGranstrom** [comment]
An alternative Vindaar proposed if you think this PR doesn't add enough new: https://discord.com/channels/371759389889003530/755344160592101389/856121013595275275

TL;DR: the `block: nbCode: ` approach doesn't work with imports so isolated/independent code blocks can't be written at the moment without first doing the imports.

**Comment by pietroppeter** [comment]
> Seems like a common use case that could use an overload, don't you agree?

I think I would keep using the `nbCode` wrapped in `block` explicitly, it would be anyway clearer for me to use (and if I really use it that often I would create a template for that).

For the use case of having imports in a scoped block, I think the answer will come when adding API to create and run a temporary file with the given code content  (which could span one or more blocks), along the lines of #34.

Regarding adding this, we could, but I would change two things:
* rename `nbCodeBlock` to `nbCodeInBlock`
* change implementation to the template one without changing signature of standard `nbCode`:

```nim
template nbCodeInBlock(body: untyped) =
  block:
    nbCode:
      body
```
this implementation does something a bit different, but for the use case it is actually the same.

**Comment by Vindaar** [comment]
Yes, this is what I ended up using here:

https://github.com/SciNim/getting-started/pull/22

I think it would be useful to have so we don't have to define it in every file we want to use it. I'll change the implementation to this.
